File: backend/src/main/java/com/placetrack/backend/service/ProfileService.java
package com.placetrack.backend.service;

import com.placetrack.backend.dto.CertificationRequest;
import com.placetrack.backend.dto.InternshipRequest;
import com.placetrack.backend.dto.ProfileResponse;
import com.placetrack.backend.dto.ProfileUpdateRequest;
import com.placetrack.backend.dto.ProjectRequest;
import com.placetrack.backend.dto.ResumeDownload;
import com.placetrack.backend.dto.SkillRequest;
import com.placetrack.backend.entity.User;
import org.springframework.web.multipart.MultipartFile;

/**
 * Placement-profile features for the logged-in student account:
 * basics + education, skills, projects, certifications, internships and resume.
 * Students can only ever access their own profile (resolved from the session token).
 */
public interface ProfileService {

    ProfileResponse getMyProfile(User user);

    ProfileResponse updateMyProfile(User user, ProfileUpdateRequest request);

    // Skills
    ProfileResponse addSkill(User user, SkillRequest request);

    ProfileResponse updateSkill(User user, Long skillId, SkillRequest request);

    ProfileResponse deleteSkill(User user, Long skillId);

    // Projects
    ProfileResponse addProject(User user, ProjectRequest request);

    ProfileResponse updateProject(User user, Long projectId, ProjectRequest request);

    ProfileResponse deleteProject(User user, Long projectId);

    // Certifications
    ProfileResponse addCertification(User user, CertificationRequest request);

    ProfileResponse updateCertification(User user, Long certificationId, CertificationRequest request);

    ProfileResponse deleteCertification(User user, Long certificationId);

    // Internships
    ProfileResponse addInternship(User user, InternshipRequest request);

    ProfileResponse updateInternship(User user, Long internshipId, InternshipRequest request);

    ProfileResponse deleteInternship(User user, Long internshipId);

    // Resume
    ProfileResponse uploadResume(User user, MultipartFile file);

    ResumeDownload downloadResume(User user);

    ProfileResponse deleteResume(User user);
}

File: frontend/src/services/profileService.js
import api from './api';

/**
 * Placement profile of the logged-in account (/api/profile).
 * Every endpoint is scoped to the caller's own profile via the X-Auth-Token
 * header that the shared axios instance attaches automatically.
 * All mutations return the full, freshly recomputed ProfileResponse.
 */
export const profileService = {
  // ===== Profile basics =====
  getProfile: () => api.get('/profile'),
  updateProfile: (data) => api.put('/profile', data),

  // ===== Resume =====
  uploadResume: (file) => {
    const formData = new FormData();
    formData.append('file', file);
    return api.post('/profile/resume', formData, {
      headers: { 'Content-Type': 'multipart/form-data' },
    });
  },
  downloadResume: () => api.get('/profile/resume', { responseType: 'blob' }),
  deleteResume: () => api.delete('/profile/resume'),

  // ===== Skills =====
  addSkill: (data) => api.post('/profile/skills', data),
  updateSkill: (id, data) => api.put(`/profile/skills/${id}`, data),
  deleteSkill: (id) => api.delete(`/profile/skills/${id}`),

  // ===== Projects =====
  addProject: (data) => api.post('/profile/projects', data),
  updateProject: (id, data) => api.put(`/profile/projects/${id}`, data),
  deleteProject: (id) => api.delete(`/profile/projects/${id}`),

  // ===== Certifications =====
  addCertification: (data) => api.post('/profile/certifications', data),
  updateCertification: (id, data) => api.put(`/profile/certifications/${id}`, data),
  deleteCertification: (id) => api.delete(`/profile/certifications/${id}`),

  // ===== Internships =====
  addInternship: (data) => api.post('/profile/internships', data),
  updateInternship: (id, data) => api.put(`/profile/internships/${id}`, data),
  deleteInternship: (id) => api.delete(`/profile/internships/${id}`),
};

export default profileService;
